### PR TITLE
Move to debian-slim base image, update tool versions, reduce layers, …

### PR DESCRIPTION
**Pull request by rtluckie**
…misc fixes

- move to debian:9.5-slim base image
- remove pull from up cmd to allow use of custom locally built images
- update git version
- update terraform version
- update minikube version
- update kubectl version
- update kops version
- add kops 1.10.0 as default
- add helm-2.10.0 and set as default
- update gomplate version
- update drone-cli version
- update git version
- update dep version
- fix vim build
- consolidate apt install and add `no-install-recommends`
- consolidate go get install into single layer
- restructure ./files to mirror target structure
- remove unnecessary sudo calls
- fix curl calls

### CODE REVIEW

**Comment by rtluckie** [comment]
reduced overall image size by ~10%, reduced layer count from 45 -> 9.

File: Dockerfile.debian [hunk 1]
```diff
@@ -59,221 +67,162 @@ RUN apt-get -y update && apt-get -y install \
...
-RUN cd /tmp && curl -L https://sourceforge.net/projects/zsh/files/zsh/5.5.1/zsh-5.5.1.tar.gz/download | tar xz && \
-    cd zsh-* && \
-    ./configure --with-tcsetpgrp --prefix=/usr/local && make && sudo make install && \
-    sudo echo "/usr/local/bin/zsh" >> /etc/shells && \
-    cd .. && rm -fr zsh-*
-
+# We sometimes install or build from source to get specific versions.
+
+RUN echo "Install/build apps that are not provided by the OS packages." && \
+    # git (needs to build first as a dependency) && \
```
Comment (drekle) [inline Dockerfile.debian]: A multi-stage build might allow us to keep the layers while squashing the image.  Layers like this are great in that they create small containers, however defeat the purpose of layers.

Comment (rtluckie) [reply Dockerfile.debian]: @drekle good call. I keep forgeting about multi-stage builds

File: Dockerfile.debian [hunk 2]
```diff
@@ -59,221 +67,162 @@ RUN apt-get -y update && apt-get -y install \
...
+        cd /tmp && curl -sSL https://github.com/git/git/archive/v2.18.0.tar.gz | tar xz && \
+        cd git-* && make configure && ./configure --prefix=/usr/local && make && make install && \
+        cd .. && rm -fr git-* && \
+    # dep && \
+        curl -sSLo dep https://github.com/golang/dep/releases/download/v0.5.0/dep-linux-amd64 && \
```
Comment (drekle) [inline Dockerfile.debian]: Build args might be neat for versions.
https://docs.docker.com/engine/reference/builder/#arg

File: Dockerfile.debian (hunk 1 above)
Comment (dcwangmit01) [reply Dockerfile.debian]: That's great insight.

**Pull request merged**